rules/merge-on-green.ts

```diff
@@ -0,0 +1,86 @@
+import { danger, peril } from "danger"
+import * as octokit from "@octokit/rest"
+
+const checkPRCondidationsAndMerge = async ({
+  number,
+  owner,
+  repo,
+}: {
+  number: number
+  owner: string
+  repo: string
+}) => {
+  // we need to check if "bot: merge on green" label is applied and PR is mergeable (checks are green and have approval)
+  const pr = await danger.github.api.pullRequests.get({ number, owner, repo })
+
+  const isMergeButtonGreen = pr.data.mergeable_state === `clean`
+
+  const hasMergeOnGreenLabel = pr.data.labels.some(
+    label => label.name === `bot: merge on green`
+  )
+
+  if (isMergeButtonGreen && hasMergeOnGreenLabel) {
+    const userAuthedAPI = new octokit()
+    userAuthedAPI.authenticate({
+      type: "token",
+      token: peril.env.GITHUB_ACCESS_TOKEN,
+    })
+    await userAuthedAPI.pullRequests.merge({
+      merge_method: `squash`,
+      commit_title: `${pr.data.title} (#${number})`,
+      number,
+      owner,
+      repo,
+    })
+  }
+}
+
+export const mergeOnGreen = async () => {
+  try {
+    if (danger.github.action === `completed` && danger.github.check_suite) {
+      // this is for check_suite.completed
+
+      // search returns first 100 results, we are not handling pagination right now
+      // because it's unlikely to get more 100 results for given sha
```

Comment:
For future reference: 

We initially tried using the `status` event but it wasn't being sent as a web hook into peril. We therefore find the pull request including this commit SHA.

Reply:
I'd recommend switching this out:

```
    const userAuthedAPI = new octokit()
	    userAuthedAPI.authenticate({
	      type: "token",
	      token: peril.env.GITHUB_ACCESS_TOKEN,
	    })
	    await userAuthedAPI.pullRequests.merge({
	      merge_method: `squash`,
	      commit_title: `${pr.data.title} (#${number})`,
	      number,
	      owner,
	      repo,
	    })
	  }
	```

and making it just 

```
danger.github.api.pullRequests.merge
```

and then you can remove that access token